### PR TITLE
Add support for deployment triggers to `prefect deploy`

### LINKED ISSUE #10046: prefect.yaml should support deployment triggers
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently deployment triggers are only supported on the old-style deployment specifications.

### Describe the proposed behavior

Deployment triggers should be supported in the new-style specifications.

### Example Use

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bunchesofdonald**
This adds support for deployment triggers to the `prefect deploy` command. Defining triggers in the prefect.yaml will result in automations being created in Prefect Cloud. A warning will be shown if you have triggers defined but are not using Cloud. 

Closes #10046 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: src/prefect/cli/deploy.py [hunk 1]
```diff
@@ -428,6 +431,22 @@ async def _run_single_deploy(
             console=app.console, client=client
         )
 
+    triggers: List[DeploymentTrigger] = []
+    trigger_specs = deploy_config.get("triggers")
+    if trigger_specs:
+        triggers = _initialize_deployment_triggers(deployment_name, trigger_specs)
+        if client.server_type != ServerType.CLOUD:
```
Comment (serinamarie) [inline src/prefect/cli/deploy.py:438]: why is it that we don't check the serve type prior to initializing deployment triggers?

File: src/prefect/cli/deploy.py [hunk 2]
```diff
@@ -428,6 +431,22 @@ async def _run_single_deploy(
...
+            app.console.print(
+                Panel(
+                    (
+                        "Deployment triggers are only supported on "
+                        "Prefect Cloud. Any triggers defined for the deployment will "
+                        "not be created."
+                    ),
+                ),
+                style="yellow",
+            )
```
Comment (serinamarie) [inline src/prefect/cli/deploy.py:448]: should we test for this output?

File: src/prefect/cli/deploy.py (hunk 1 above)
Comment (bunchesofdonald) [reply src/prefect/cli/deploy.py:438]: My thinking was that we should still validate the triggers property even if we aren't going to, ultimately, create them.

File: src/prefect/cli/deploy.py (hunk 2 above)
Comment (bunchesofdonald) [reply src/prefect/cli/deploy.py:448]: Good call.

**Review by desertaxle — APPROVED** [review]
LGTM! 

It'd be great to have at least one test with a full `prefect deploy` run to ensure that the CLI makes the expected API call(s) when triggers are configured.

**Comment by bunchesofdonald** [comment]
> LGTM!
> 
> It'd be great to have at least one test with a full `prefect deploy` run to ensure that the CLI makes the expected API call(s) when triggers are configured.

It would, but I couldn't figure out how to make that work when we have to mock calls to cloud, but `invoke_and_assert` has to be called in a thread. If you have ideas I'd love to hear them.

File: src/prefect/cli/deploy.py (hunk 1 above)
Comment (serinamarie) [reply src/prefect/cli/deploy.py:438]: oh I see, it's just validating them 👍

Comment (bunchesofdonald) [reply src/prefect/cli/deploy.py:438]: Right, it's initializing the pydantic model, which also validates.

**Comment by desertaxle** [comment]
> It would, but I couldn't figure out how to make that work when we have to mock calls to cloud, but `invoke_and_assert` has to be called in a thread. If you have ideas I'd love to hear them.

I think I did something like that with `respx` in https://github.com/PrefectHQ/prefect/pull/10040. If using `respx` doesn't work here then we can forgo the extra test.

**Comment by serinamarie** [comment]
Hi @bunchesofdonald, does this issue close #10223?

**Comment by bunchesofdonald** [comment]
Hi @serinamarie, no #10223 is about allowing triggers to be defined at the top-level and not attached to any specific deployment.

**Comment by serinamarie** [comment]
Oh gotcha, this is just for specific deployments, okay.

**Pull request merged**